rs/crypto/src/sign/threshold_sig/store.rs

```diff
@@ -115,29 +115,16 @@ impl Default for ThresholdSigDataStoreImpl {
 impl ThresholdSigDataStoreImpl {
```

Comment:
We could add something like `const _: () = assert!(ThresholdSigDataStoreImpl::CAPACITY_PER_TAG_OR_KEY > 0, "The maximum size per tag must be at least 1");` after
```rust
impl ThresholdSigDataStoreImpl {
    // ...
}
```
and maybe even remove the test.

You can test it. If you set `const _: () = assert!(ThresholdSigDataStoreImpl::CAPACITY_PER_TAG_OR_KEY > 10, "The maximum size per tag must be at least 1");`, it won't compile and the IDE will show you the error message.

Reply:
> and maybe even remove the test

What the assertion you propose assert, and what the test `should_have_capacity_per_tag_or_key_of_9` tests are two different things though. The former ensures that the value is `>0`, while the test tests that the value is `=9`.

I'd say if we make the assertion assert `=9`, then we can remove the test, no? Then the assertion basically is the test, but at compile time.

Is this also your view?

Reply:
In general yes, I'm just not sure what we exactly want to assert on. If we want to say that we want that value to always be `9`, we should probably add something like "the value should be 9 because x, y, z, and it should be a conscious decision to change it". While about the assertion `>0` is quite intuitive because the capacity `== 0` just won't work IIUC.

Reply:
I adapted the code to have two assertions. Not 100% sure of why a value of exactly 9 was chosen at the time. IIRC, also slightly smaller values would be OK, but 9 was considered safe with a little overprovisioning.

In any case, the assertions feel like tests, so for a while I was contemplating to move them to `tests.rs`, but then they are only compiled/checked if the tests are compile (which may very well fine fine). Anyway, I left them in the production code.

Reply:
👍